lib/mainloop-control.c

```diff
@@ -106,10 +106,16 @@ static void
 control_connection_config(ControlConnection *cc, GString *command, gpointer user_data)
 {
   MainLoop *main_loop = (MainLoop *) user_data;
-  GlobalConfig *config = main_loop_get_current_config(main_loop);
+  GlobalConfig *config = main_loop_lock_current_config(main_loop);
   GString *result = g_string_sized_new(128);
   gchar **arguments = g_strsplit(command->str, " ", 0);
 
+  if (!config)
+    {
+      g_string_assign(result, "syslog-ng is being reloaded");
```

Comment:
hmmm... I don't really like this idea.
best would be to propagate back an error code

Reply:
well I am not keen on either. I was thinking about the error code, and my issue with that was that it would break how `syslog-ng-ctl` worked. Now if `syslog-ng-ctl` reports error code, there were an issue with the reload. As such systemd or script could use it.

Maybe a combination that syslog-ng does report back an error code or something. But `syslog-ng-ctl` on this report tries to do a reload again. But I do not like this as it still would need some limitation and in the end.